src/Symfony/Component/Security/Http/EntryPoint/FormAuthenticationEntryPoint.php

```diff
@@ -54,7 +54,7 @@ public function start(Request $request, AuthenticationException $authException =
 
             $response = $this->httpKernel->handle($subRequest, HttpKernelInterface::SUB_REQUEST);
             if (200 === $response->getStatusCode()) {
-                $response->headers->set('X-Status-Code', 401);
+                $response->setStatusCode(401);
```

Comment:
@fabpot I'm not sure about this change, in the authentication listeners found in the `Symfony\Component\Security\Http\Firewall` namespace the response returned from the `AuthenticationEntryPointInterface::start()` method is set on the response. From what I can see the `X-Status-Code` is only ever used when handling the exception in the kernel, so I don't **think** this will have any adverse impact